content/en/blog/_posts/2022-12-16-non-graceful-node-shutdown-to-beta.md

```diff
@@ -0,0 +1,80 @@
+
+## What is a Node Shutdown in Kubernetes
+
+In a Kubernetes cluster, it is possible for a node to shut down. This could happen either in a planned way or it could happen unexpectedly. You may plan for a security patch, or a kernel upgrade and need to reboot the node, or it may shut down due to preemption of VM instances. A node may also shut down due to a hardware failure or a software problem.
+
+To trigger a node shutdown, you could run a shutdown or poweroff command or physically press a button to power off a machine.
+
+A node shutdown could lead to workload failure if the node is not drained before the shutdown.
+
+In the following, we will describe what is a graceful node shutdown and what is a non-graceful node shutdown.
+
+## What is a Graceful Node Shutdown
+
+[Graceful Node Shutdown](https://kubernetes.io/blog/2021/04/21/graceful-node-shutdown-beta/) feature allows the kubelet to detect a node shutdown event, properly terminate the pods, and release resources before the actual shutdown. [Critical pods](/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/#marking-pod-as-critical) will be terminated after all the regular pods are terminated to ensure that the essential functions of an application can continue to work as long as possible.
+
+## What is a Non-Graceful Node Shutdown
+
+A Node Shutdown can be `Graceful` only if kubelet's Node Shutdown Manager can detect the node shutdown action. However, there are cases where kubelet's Node Shutdown Manager may not detect a node shutdown action. This could happen because the shutdown command does not trigger the [Inhibitor Locks](https://www.freedesktop.org/wiki/Software/systemd/inhibit) mechanism used by kubelet or because of a user error, i.e., the `ShutdownGracePeriod` and `ShutdownGracePeriodCriticalPods` are not configured correctly.
+
+When a node is shut down but not detected by kubelet's Node Shutdown Manager, it becomes a non-graceful node shutdown. Non-graceful node shutdown is a problem for stateful apps. If a node containing a pod that is part of a StatefulSet is shut down in a non-graceful way, the pod will be stuck in `Terminating` status and cannot move to a new running node. Similarly, pods that ReplicaSets create as part of a Deployment will be stuck in `Terminating` status on the shutdown node forever. However, they will come up on another running node and be stuck in the `ContainerCreating` status on the new node. As a result, the application will not function properly. If the original shutdown node comes up, the old pod will be deleted by kubelet, and the new pod will be created successfully on a different running node.
+
+## What’s new in Beta
+
+With the promotion of the Non-Graceful Node Shutdown feature to beta, the feature gate `NodeOutOfServiceVolumeDetach` is enabled by default on `kube-controller-manager` instead of being opt-in.
```

Comment:
How about:
```suggestion
For Kubernetes v1.26, volume detachment on out-of-service nodes is beta and enabled by default.
The `NodeOutOfServiceVolumeDetach`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled by default
on `kube-controller-manager` instead of being opt-in; you can still disable it if needed
(please also file an issue to explain the problem).
```